src/redux/tasks/actions/delete.ts

```diff
@@ -0,0 +1,52 @@
+import { Dispatch } from '../../RootReducer'
+import { API } from '../../../services/http'
+
+import { ITask } from '../types'
+
+export const REMOVE_TASK = 'REMOVE_TASK'
+export const REMOVE_TASK_ERROR = 'REMOVE_TASK_ERROR'
+export const REMOVE_TASK_SUCCESS = 'REMOVE_TASK_SUCCESS'
+
+export type Actions = {
+    REMOVE_TASK: {   
+        type: typeof REMOVE_TASK,
+    },
+    REMOVE_TASK_ERROR: {     
+        type: typeof REMOVE_TASK_ERROR,
+        error: string
+    },
+    REMOVE_TASK_SUCCESS: {   
+        type: typeof REMOVE_TASK_SUCCESS,
+        task: ITask
+    },
+}
+
+export const actionCreators = {
+    // --------------------------------------- //
+    //                    SYNC                 //
+    // --------------------------------------- //
+    removeTaskRequest: (): Actions[typeof REMOVE_TASK] => ({
+        type: REMOVE_TASK,
+    }),
+    removeTaskRequestError: (error: string): Actions[typeof REMOVE_TASK_ERROR] => ({
+        type: REMOVE_TASK_ERROR,
+        error
+    }),
+    removeTaskRequestSucess: (task: ITask): Actions[typeof REMOVE_TASK_SUCCESS] => ({
+        type: REMOVE_TASK_SUCCESS,
+        task
+    }),
+
+    // --------------------------------------- //
+    //                   ASYNC                 //
+    // --------------------------------------- //
+    removeBackendTask: (task: ITask) => {
+        return (dispatch: Dispatch) => {
+            dispatch(actionCreators.removeTaskRequest())
+            return API.delete('/checkitems/', task.id).then(
+                response => dispatch(actionCreators.removeTaskRequestSucess(response.task)),
```

Comment:
Why the API would send back the task that it just removed ?

Reply:
I use the exemple created for List by Alexis, so in doubt I didn't changed it. I'll ask him.

Reply:
Change response by task